superset/models/core.py

```diff
@@ -773,6 +775,10 @@ def name(self):
     def allows_subquery(self):
         return self.db_engine_spec.allows_subqueries
 
+    @property
+    def allows_cost_estimate(self):
+        return self.db_engine_spec.allows_cost_estimate
```

Comment:
with the current implementation, shouldn't this be:
`return self.cost_estimate_enabled and self.db_engine_spec.allows_cost_estimate`

Reply:
I preferred to implement that logic in the frontend, exposing the 2 separate flags.